cou/utils/juju_utils.py

```diff
@@ -383,7 +415,18 @@ async def get_applications(self) -> dict[str, Application]:
                 series=status.series,
                 subordinate_to=status.subordinate_to,
                 units={
-                    name: Unit(name, machines[unit.machine], unit.workload_version)
+                    name: Unit(
+                        name,
+                        machines[unit.machine],
+                        unit.workload_version,
+                        [
+                            SubordinateUnit(
+                                subordinate_name,
+                                unit_charm_mapping[subordinate_name],
+                            )
+                            for subordinate_name, subordinate in unit.subordinates.items()
```

Comment:
Not necessary to use `subordinate` anymore
```suggestion
                            for subordinate_unit in unit.subordinates
```